src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -1,32 +1,196 @@
 
 @RestController
 public class MigrationController {
 
-    private static final String template = "Hello, %s!";
+	private static final Logger log = LoggerFactory.getLogger(MigrationController.class);
+	
     private final AtomicLong counter = new AtomicLong();
     
     @Autowired MigrationRepository repository;
+    
+    @Autowired
+    private Environment env;
+    
+    /**
+     * get all CTools sites where user have site.upd permission 
+     * @return
+     */
+    @RequestMapping("/projects")
+    public String getProjectSites() {
+    	String rv = "";
+    	
+    	// login to CTools and get sessionId
+    	String sessionId = login_becomeuser();
+    	if (sessionId != null)
+    	{
+			// get all sites that user have permission site.upd
+			RestTemplate restTemplate = new RestTemplate();
+			// the url should be in the format of "https://server/direct/site/withPerm/.json?permission=site.upd"
+			String requestUrl = env.getProperty("ctools.direct.url") + "site/withPerm/.json?permission=site.upd&_sessionId=" + sessionId;
+			log.info(requestUrl);
+        	try
+        	{
+        		rv= restTemplate.getForObject(requestUrl, String.class);
+        	}
+        	catch (RestClientException e)
+        	{
+        		log.error(requestUrl + e.getMessage());
+        	}
+        }
+        return rv;
+    }
+    
+    /**
+     * get page information
+     * @param site_id
+     * @return
+     */
+    @RequestMapping("/projects/{site_id}")
+    public String getProjectSitePages(@PathVariable String site_id) {
+    	String rv = "";
+    	
+    	// login to CTools and get sessionId
+		String sessionId = login_becomeuser();
+		if (sessionId != null)
+		{	
+        	//3. get all sites that user have permission site.upd
+        	RestTemplate restTemplate = new RestTemplate();
+        	// the url should be in the format of "https://server/direct/site/SITE_ID.json"
+        	String requestUrl = env.getProperty("ctools.direct.url") + "site/" + site_id + "/pages.json?_sessionId=" + sessionId;
+        	log.info(requestUrl);
+        	
+        	try
+        	{
+        		rv= restTemplate.getForObject(requestUrl, String.class);
+        	}
+			catch (RestClientException e)
+			{
+				log.error(requestUrl + e.getMessage());
+			}
+        }
+		return rv;
+    }
 
+	/**
+	 * login into CTools and become user with sessionId
+	 */
+	private String login_becomeuser() {
+		// return the session id after login
+		String sessionId = "";
+		
+		// here is the CTools integration prior to CoSign integration ( read session user information from configuration file)
+    	// 1. create a session based on user id and password
+    	RestTemplate restTemplate = new RestTemplate();
+    	// the url should be in the format of "https://server/direct/session?_username=USERNAME&_password=PASSWORD"
+    	String requestUrl = env.getProperty("ctools.direct.url") + "session?_username=" + env.getProperty("username") + "&_password=" + env.getProperty("password");
```

Comment:
SpringApplication will load properties from application.properties files in the following locations and add them to the Spring Environment:

http://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html